lib/util/chat/pages/chat_page.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import '../../../account/models/profile.dart';
+import '../../profiles/profile_widget.dart';
+import '../../supabase.dart';
+import '../chat_bubble.dart';
+import '../message_bar.dart';
+import '../models/message.dart';
+
+class ChatPage extends StatefulWidget {
+  final int rideId;
+  final Profile profile;
+  final bool chatExists;
+
+  const ChatPage({
+    rideId,
+    required this.profile,
+    this.chatExists = true,
+    super.key,
+  }) : rideId = rideId ?? -1;
+
+  @override
+  State<ChatPage> createState() => _ChatPageState();
+}
+
+class _ChatPageState extends State<ChatPage> {
+  late Stream<List<Message>> _messagesStream;
+
+  @override
+  void initState() {
+    if (widget.chatExists) {
```

Comment:
this is super weird. I just tried to start a chat on a cancelled ride. I can send messages, but nothing happens. But shouldn't you see the old messages when it is cancelled?

Only if it's not approved yet, the behaviour should be different. But in that case sending a message should be impossible, it should just show a placeholder message

Reply:
okay that's weird.

Reply:
okay i'm not sure what you mean. what i can add is that the chat is only shown when there are messages, if the ride is not active anymore